### ISSUE TITLE
[Question] Does BOLT work with upstream llvm/project?

### ISSUE THREAD

**Issue by kiranchandramohan** [facebookincubator/BOLT#176]
Does BOLT work with upstream llvm/project?
Is there a version that works with the monorepo?
The Optimizing Clang tutorial works with llvm-7, is there a later version of LLVM that is supported?
https://github.com/facebookincubator/BOLT/blob/master/docs/OptimizingClang.md

Thanks in advance.

**Comment by Sameeranjoshi** [comment]
> Does BOLT work with upstream llvm/project?
> Is there a version that works with the monorepo?

Yes, initially the project was tightly coupled with multi-repo of LLVM, recently after announcement[3] into the LLVM community the effort had started to move to mono-repo version and even has been accepted to be merged into LLVM project soon.
You can try the `rebased` branch, it's based out of mono-repo.
https://github.com/facebookincubator/BOLT/tree/rebased

> The Optimizing Clang tutorial works with llvm-7, is there a later version of LLVM that is supported?
> https://github.com/facebookincubator/BOLT/blob/master/docs/OptimizingClang.md

I tried with clang-13[1] but due to machine issue(for LBR support Intel machine is recommended, which was missing for me, maybe you can give it a try if you have LBR supported machine.) I couldn't take it forward.
And incase you are trying to play with BOLT to see how to have an end to end run and get an application optimized you can also try this[2] great blog by the author.

> 
> Thanks in advance.

[1] https://github.com/facebookincubator/BOLT/issues/146
[2] https://aaupov.github.io/blog/2020/10/08/bolt-nodejs
[3] https://lists.llvm.org/pipermail/llvm-dev/2020-October/145902.html

**Comment by kiranchandramohan** [comment]
Hey @Sameeranjoshi, great to see you here. Thanks for the reply. I will give it a try later this week. My use case is to build a fast version of Clang to build Flang. Using and LTO build helped, hoping that BOLT can improve on that. Preferably, I want to test on AArch64 machines.
